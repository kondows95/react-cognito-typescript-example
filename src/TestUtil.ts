import {ReactElement} from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
export default class TestUtil
{
  public render: RenderResult;
  constructor(component:ReactElement) {
    this.render = render(component);
  }

  setValue(testId: string, value: string | number) {
    const elem = this.render.getByTestId(testId);
    fireEvent.change(elem, {target: {value: value}})
  }

  click(testId: string) {
    const elem = this.render.getByTestId(testId);
    fireEvent.click(elem);
  }

  getAll(testId: string) {
    return this.render.getAllByTestId(testId);
  }

  get(testId: string) {
    return this.render.getByTestId(testId);
  }
}