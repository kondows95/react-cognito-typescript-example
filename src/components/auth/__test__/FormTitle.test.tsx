import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import FormTitle from '../FormTitle';
import TestUtil from '../../../TestUtil'; 

describe("FormTitle Component",()=>{
  let util: TestUtil;
  beforeEach(() => {
    util= new TestUtil(
      <FormTitle children={"Form Title"}/>
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });

  it('Check text', () => {
    expect(util.render.container).toHaveTextContent('Form Title'); 
  });
});