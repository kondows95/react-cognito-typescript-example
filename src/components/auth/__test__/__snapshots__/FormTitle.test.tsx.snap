// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTitle Component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
          >
            <path
              d="M0 0h24v24H0V0z"
            />
            <path
              d="M0 0h24v24H0V0z"
              opacity=".87"
            />
          </g>
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-14"
    >
      Form Title
    </div>
  </div>
</DocumentFragment>
`;
